middle_end/flambda/types/inlining/inlining_arguments.ml

```diff
@@ -14,65 +14,158 @@
 
 [@@@ocaml.warning "+a-4-30-40-41-42"]
 
-type t =
-  | Unknown
-  | O1
-  | O2
-  | O3
-  | Custom of { max_inlining_depth : int }
-
-let unknown = Unknown
-
-let print ppf = function
-  | Unknown -> Format.fprintf ppf "Unknown"
-  | O1 -> Format.fprintf ppf "O1"
-  | O2 -> Format.fprintf ppf "O2"
-  | O3 -> Format.fprintf ppf "O3"
-  | Custom t ->
+module Args = struct
+  type t = {
+    max_inlining_depth : int;
+    call_cost : float;
+    alloc_cost : float;
+    prim_cost : float;
+    branch_cost : float;
+    indirect_cost : float;
+    small_function_size : int;
+    large_function_size : int;
+    threshold : float;
+  }
+
+  let print ppf t =
     Format.fprintf ppf
-      "@[<hov 1>(max_inlining_depth %d)@]"
+      "@[<hov 1>(@ \
+       @[<hov 1]>(max_inlining_depth@ %d)@]@ \
+       @[<hov 1]>(call_cost@ %f)@]@ \
+       @[<hov 1]>(alloc_cost@ %f)@]@ \
+       @[<hov 1]>(prim_cost@ %f)@]@ \
+       @[<hov 1]>(branch_cost@ %f)@]@ \
+       @[<hov 1]>(indirect_cost@ %f)@]@ \
+       @[<hov 1]>(small_function_size@ %d)@]@ \
+       @[<hov 1]>(large_function_size@ %d)@]@ \
+       @[<hov 1]>(threshold@ %f)@]\
+      )@]"
       t.max_inlining_depth
+      t.call_cost
+      t.alloc_cost
+      t.prim_cost
+      t.branch_cost
+      t.indirect_cost
+      t.small_function_size
+      t.large_function_size
+      t.threshold
+
+  let equal t1 t2 =
```

Comment:
Please use exhaustive record pattern matches instead of `equal t1 t2` so we get a compiler error if we add a new field and forget to update this function.  Same for `<=` and `meet`.

Reply:
Done for all of them.